api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -426,6 +454,88 @@ public void setHttpOnly(boolean isHttpOnly) {
      * @since Servlet 3.0
      */
     public boolean isHttpOnly() {
-        return isHttpOnly;
+        return Boolean.parseBoolean(getAttribute(HTTP_ONLY));
+    }
+
+    /**
+     * Sets the value of the cookie attribute associated with the given name.
+     * 
+     * <p>
+     * This should sync to any predefined attribute for which already a getter/setter pair exist in the current version,
+     * except for <code>version</code>. E.g. when <code>cookie.setAttribute("domain", domain)</code> is invoked, then
+     * <code>cookie.getDomain()</code> should return exactly that value, and vice versa.
+     * 
+     * @param name the name of the cookie attribute to set the value for, case insensitive
+     * 
+     * @param value the value of the cookie attribute associated with the given name, can be {@code null}
+     *
+     * @throws IllegalArgumentException if the cookie name is null or empty or contains any illegal characters (for example,
+     * a comma, space, or semicolon) or matches a token reserved for use by the cookie protocol
+     * 
+     * @since Servlet 5.1
+     */
+    public void setAttribute(String name, String value) {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException(createErrorMessage("err.cookie_attribute_name_blank"));
+        }
+
+        if (containsReservedToken(name)) {
+            throw new IllegalArgumentException(createErrorMessage("err.cookie_attribute_name_is_token", name));
+        }
+
+        if (MAX_AGE.equalsIgnoreCase(name) && value != null)
+            Long.parseLong(value);
```

Comment:
This is not good for forward compatibility.

Reply:
I don't think I can just remove it, as then we have to deal with what does getMaxAge() do if the value is not a number.   Either we throw here or we throw later or we treat not a number as -1.

My thought is that allowing illegal values will just make our life more difficult in future.  So if we throw, then we should throw on the setter not the getter (else we are victim blaming!).

Reply:
Imagine we end up with 10 new non-string attributes 10 year later. Should setAttribute() have possibly 10 different fallbacks/checks for all of them and have them all documented in its javadoc? This is just code smell.

Reply:
I think yes.  Sure it is a PITA, but it is far better to check arguments than to allow illegal values into the map.     I have changed so that rather than duplicate the checking, it now just calls `setMaxAge`

Reply:
I disagree.

Reply:
Any other opinions on this? @markt-asf @stuartwdouglas 

I much prefer fail fast.   Otherwise the exception will be thrown from the getting inside addSetCookie and it will have to throw back to the application.   Much better to throw to the application when the Cookie is being created.

Reply:
Nope. The whole idea of forward compatibility of `attributes` is that you should use `cookie.getAttributes().get("Max-Age")` instead to obtain it (as well as all others). Otherwise `SameSite` is going to bite back when it's going to be implemented as an `enum` and HTTP guys add a new value.

Reply:
But then something needs to check the value, what we risk sending invalid set-cookies.

I guess new fields unsupported by the api will be checked and thrown from addSetCookie.... Hmmm so maybe ok to do it all there.  I'm convincible... but I'd like to hear from others

Reply:
I am +1 on Greg's approach. We should not be allowing invalid values into the map.

Reply:
This could be implemented as a separate cookie-attribute-validator which could be applied at a particular time, either by a client or by a server, instead of placing these checks in `setAttribute` directly.

Reply:
Sorry se this after the merge. I think calling the setter avoids duplicating the validation logic simply enough